parsl/app/watcher.py

```diff
@@ -0,0 +1,96 @@
+"""This module provides the bash_watch_app function. This function wraps a bash_app and captures
```

Comment:
is this anything bash_app specific? (vs wrapping a python app?)

Reply:
It is not bash_app specific and could be used by a python_app. However, it is less efficient, prone to potential issues (as you mentioned above) and would be redundant (as it attempts to do what the python_app will do in the background anyway).